### PR TITLE
Exclude sub-project files from file and editor watchers

### PR DESCRIPTION
**Pull request by mattseddon**
This PR fixes some sub-project-related issues with our file system and editor watchers. We were previously not correctly excluding sub-project files from their parent project's watchers.

### CODE REVIEW

File: extension/src/pipeline/data.ts [hunk 1]
```diff
@@ -27,8 +37,10 @@ export class PipelineData extends BaseData<{
     ])
 
     const dvcYamlsDirs = new Set<string>()
+    this.collectedFiles = []
     for (const file of fileList) {
-      if (file.startsWith(this.dvcRoot)) {
+      if (isPathInProject(file, this.dvcRoot, this.subProjects)) {
+        this.collectedFiles.push(file)
```
Comment (mattseddon) [inline extension/src/pipeline/data.ts:43]: [B] Not having `collectedFiles` for this class was actually a bug.

Comment (mattseddon) [reply extension/src/pipeline/data.ts:43]: discovered by writing a test for this functionality

File: extension/src/fileSystem/index.test.ts [hunk 2]
```diff
@@ -408,3 +409,51 @@ describe('findOrCreateDvcYamlFile', () => {
...
   })
 })
+
+describe('isPathInProject', () => {
+  it('should return true if the path is in the project', () => {
+    const path = join(dvcDemoPath, 'dvc.yaml')
+    const dvcRoot = dvcDemoPath
+    const subProjects: string[] = []
+    expect(isPathInProject(path, dvcRoot, subProjects)).toBe(true)
+  })
```
Comment (julieg18) [inline extension/src/fileSystem/index.test.ts:419]: Looks like we're missing a new line here:
```suggestion
  })
  
```

**Review by julieg18 — APPROVED** [review]

File: extension/src/fileSystem/index.test.ts (hunk 2 above)
Comment (mattseddon) [reply extension/src/fileSystem/index.test.ts:419]: I wonder if there is a prettier rule for this

Comment (mattseddon) [reply extension/src/fileSystem/index.test.ts:419]: maybe this https://eslint.org/docs/latest/rules/padding-line-between-statements

**Pull request merged**